docs/HowTo/develop-subtopologies.md

```diff
@@ -0,0 +1,345 @@
+```
+
+Then modify the current `PingEntires` namespace call to be surrounded by GlobalDefs:
+
+```cpp
+namespace GlobalDefs {
+    namespace PingEntries {
+        namespace blockDrv {
+            enum { WARN = 3, FATAL = 5 };
+        }
+...
+```
+
+Then, we want to tell our MainDeployment's topology to import and use our unified topology file from our subtopology. While we're here, we should also hook up the `CycleOut` port of our main deployment's rate group driver to the `CycleIn` port of our subtopology's rate group. Ensure that the rate group driver can have enough output wires going from the output port. We head over to the `topology.fpp` file, and include:
+
+```cpp
+...
+topology MainDeployment {
+    import MySubtopology.MySubtopology
+    ...
+
+    connections RateGroups{
+        ...
+        rateGroupDriver.CycleOut[3] -> MySubtopology.rateGroup.CycleIn // you'll notice that the syntax here is <Namespace>.<instance>
+        ...
+    }
+}
+...
+```
+
+At this point, we want to now call our subtopology's configure/start/teardown functions within the corresponding functions of our `MainDeployment` topology. So, in `MainDeploymentTopology.cpp`:
+
+```cpp
+// at the top, include our topology.hpp
+#include <MySubtopology/MySubtopologyTopology.hpp>
+
+...
+// MODIFY this line to include the 4th divider
+// Svc::RateGroupDriver::DividerSet rateGroupDivisorsSet{{{1, 0}, {2, 0}, {4, 0}, {1, 0}}};
+
+void configureTopology() {
+    ...
+    MySubtopology::TopologyState = myState;
+    myState.clockRate = 1; // we set our clock rate to whatever standard we want
+    MySubtopology::configureTopology(myState);
+}
+
+...
+
+void setupTopology(const TopologyState& state){
+    ...
+    MySubtopology::startTopology({});
+}
+
+...
+
+void teardownTopology(const TopologyState& state){
+    ...
+    MySubtopology::teardownTopology({})
+}
+```
+
+Lastly, since our RNG component has some telemetry, we need to include (or ignore) these channels within the `Packets.xml` file in this folder. As with any other component that is added to a deployment, you use the same syntax with the name of the instance followed by the name of the telemetry channel. For example:
+
+```xml
+<channel name="rng.RNGValue"/> <!-- based on our instance name -->
+```
+
+Now go ahead and run and build your deployment, and you should see that you have a built deployment that uses a subtopology.
+
+# Conclusion
+
+This how-to guide has walked through the development of a subtopology. Deployments can include multiple different subtopologies, and thus this feature truly paves the way for making F Prime more accessible to quick prototyping. 
+
+If you'd like to see an example of how subtopologies are used within an actual project, please reference this repository: [RNG Library - An Example of Subtopologies within F'](https://github.com/mosa11aei/fprime-rngLibrary). This project uses the [example scenario](#example-scenario) from this guide.
```

Comment:
I would advertise this reference project at the very top of the guide so users can follow along with it? Seems very helpful
Also do we want to migrate this repo to e.g. `fprime-community/fprime-subtopology-reference` or something like this? This is awesome!